platforms/core-configuration/kotlin-dsl-tooling-builders/src/main/kotlin/org/gradle/kotlin/dsl/tooling/builders/KotlinLibSources.kt

```diff
@@ -16,24 +16,158 @@
 
 package org.gradle.kotlin.dsl.tooling.builders
 
+import org.gradle.api.Project
 import org.gradle.api.artifacts.component.ComponentIdentifier
 import org.gradle.api.artifacts.component.ModuleComponentIdentifier
 import org.gradle.api.artifacts.dsl.DependencyHandler
 import org.gradle.api.artifacts.query.ArtifactResolutionQuery
 import org.gradle.api.artifacts.result.ResolvedArtifactResult
-
 import org.gradle.api.initialization.dsl.ScriptHandler
 import org.gradle.api.initialization.dsl.ScriptHandler.CLASSPATH_CONFIGURATION
-
 import org.gradle.internal.classpath.ClassPath
 import org.gradle.internal.classpath.DefaultClassPath
-
 import org.gradle.jvm.JvmLibrary
+import org.gradle.kotlin.dsl.*
+import org.gradle.language.base.artifact.SourcesArtifact
+import org.gradle.tooling.provider.model.ParameterizedToolingModelBuilder
+import org.gradle.tooling.provider.model.internal.IntermediateToolingModelProvider
 
-import org.gradle.kotlin.dsl.embeddedKotlinVersion
-import org.gradle.kotlin.dsl.get
 
-import org.gradle.language.base.artifact.SourcesArtifact
+internal
+interface ScriptSourcePathRequestParameter {
+    val skipDependencies: Set<ComponentIdentifier>
+    val resolveKotlinLibSources: Boolean
+}
+
+
+internal
+data class DefaultScriptSourcePathRequestParameter(
+    override val skipDependencies: Set<ComponentIdentifier>,
+    override val resolveKotlinLibSources: Boolean
+) : ScriptSourcePathRequestParameter
+
+
+internal
+data class ScriptSourcePathResolutionResult(
```

Comment:
💭 Why you decided to use **Source**Path name instead of ClassPath for these entities?

Reply:
Because it sources jars of the dependencies. It's a shorthand for "sources classpath"